### PR TITLE
Update VersionedLayerClient constructor

### PR DESCRIPTION
**Pull request by AndriyShevchuk**
* Update VersionedLayerClient constructor,
* Add getLatestVersion method,
* Update tests

Resolves: OLPEDGE-922
Signed-off-by: ashevchu <ext-andrii.shevchuk@here.com>

### CODE REVIEW

File: @here/olp-sdk-dataservice-read/test/VersionLayerClient.test.ts [hunk 1]
```diff
@@ -512,6 +518,7 @@ function createMockDownloadManager(): DownloadManager {
     const downloadMgr = sinon.createStubInstance(DataStoreDownloadManager);
 
     downloadMgr.download.callsFake((url: string, init?: RequestInit) => {
+        console.log(url);
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/test/VersionLayerClient.test.ts]: Please remove

File: @here/olp-sdk-dataservice-read/lib/VersionLayerClient.ts [hunk 2]
```diff
@@ -86,11 +64,30 @@ export class VersionLayerClient {
...
+    constructor(catalogHrn: HRN, layerId: string, context: DataStoreContext) {
+        this.context = context;
+        this.hrn = catalogHrn.toString();
+        this.layerId = layerId;
+    }
+
+    /**
+     * Gets the latest available catalog version what can be used as latest layer version
+     */
+    async getLatestVersion(): Promise<MetadataApi.VersionResponse> {
```
Comment (OleksiiZubko) [inline @here/olp-sdk-dataservice-read/lib/VersionLayerClient.ts]: it's could be private

File: @here/olp-sdk-dataservice-read/lib/VersionLayerClient.ts [hunk 3]
```diff
@@ -31,38 +31,16 @@ import {
     MetadataApi,
     QueryApi
 } from "@here/olp-sdk-dataservice-api";
+import { HRN } from "@here/olp-sdk-dataservice-read";
```
[the remark this answers is not in the document]
Comment (AndriyShevchuk) [reply @here/olp-sdk-dataservice-read/lib/VersionLayerClient.ts]: fixed

File: @here/olp-sdk-dataservice-read/test/VersionLayerClient.test.ts (hunk 1 above)
Comment (AndriyShevchuk) [reply @here/olp-sdk-dataservice-read/test/VersionLayerClient.test.ts]: Removed

File: @here/olp-sdk-dataservice-read/lib/VersionLayerClient.ts (hunk 2 above)
Comment (AndriyShevchuk) [reply @here/olp-sdk-dataservice-read/lib/VersionLayerClient.ts]: done

File: @here/olp-sdk-dataservice-read/lib/VersionLayerClient.ts [hunk 4]
```diff
@@ -86,11 +64,30 @@ export class VersionLayerClient {
...
+    async getLatestVersion(): Promise<MetadataApi.VersionResponse> {
+        const builder = await this.getRequestBuilder("metadata").catch(error =>
+            Promise.reject(error)
+        );
+        return MetadataApi.latestVersion(builder, {
+            startVersion: -1
+        }).catch(async (error: Response) =>
+            Promise.reject(
+                new Error(
+                    `Blob Service error: HTTP ${
```
[the remark this answers is not in the document]
Comment (AndriyShevchuk) [reply @here/olp-sdk-dataservice-read/lib/VersionLayerClient.ts]: Fixed

**Pull request merged**